gem/node.py

```diff
@@ -36,14 +37,17 @@ def _cons_args(self, children):
 
         Internally used utility function.
         """
-        front_args = [getattr(self, name) for name in self.__front__]
-        back_args = [getattr(self, name) for name in self.__back__]
+        front_args = (getattr(self, name) for name in self.__front__)
+        back_args = (getattr(self, name) for name in self.__back__)
 
-        return tuple(front_args) + tuple(children) + tuple(back_args)
+        return (*front_args, *children, *back_args)
+
+    def _arguments(self):
```

Comment:
Should this be a (cached?) property?

Reply:
It should

Reply:
I cannot add `"_arguments"` to `__slots__`: 
```
ValueError: '_arguments' in __slots__ conflicts with class variable
```

Reply:
But don't make this a `cached_property` anyway. Although nodes are immutable, the eager dagification in `__eq__` rewrites `self.children` to point to `other.children` when the two objects compare equal. Stashing the `_arguments` as a cached property would destroy this.

Reply:
So `property` is fine?